tests/trx_generator/trx_provider.cpp

```diff
@@ -50,28 +53,64 @@ namespace eosio::testing {
       _p2p_socket.send(boost::asio::buffer(*msg));
    }
 
-   p2p_trx_provider::p2p_trx_provider(const provider_base_config& provider_config) :
-      _peer_connection(provider_config) {
+   void http_connection::connect() {}
 
+   void http_connection::disconnect() {
+      int max    = 30;
+      int waited = 0;
+      while (_sent != _acknowledged && waited < max) {
+         ilog("http_connection::disconnect waiting on ack - sent ${s} | acked ${a} | waited ${w}",
+              ("s", _sent)("a", _acknowledged)("w", waited));
+         sleep(1);
+         ++waited;
+      }
```

Comment:
Seems like you should `elog` if waited == max.

Reply:
Addressed: [f342f5e](https://github.com/AntelopeIO/leap/pull/1138/commits/f342f5e3cace6bd7db29e548640fc9a23fe53d43)

Reply:
I expected that `if` outside the loop.

Reply:
That is probably better, yes.  Done. [d64ccd7](https://github.com/AntelopeIO/leap/pull/1138/commits/d64ccd7d40d2d94d97654d578903b6e61d993dc8)